### PR TITLE
Migrate Sort (TopK-1)

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Create template plugin reference tests for target operation: TopK-1*
 - *Remove ngraph backend files*
 - *Update CMakeLists.txt*

### Tickets:
 - *64054*

### CODE REVIEW

**Comment by steve-y** [comment]
@iefode @ilyachur This PR does not include `topk_mode_sort_order` of `topk_backend` test using `TYPED_TEST_P` in topk.in.cpp. Should it be implemented in op_reference test?

**Comment by steve-y** [comment]
Window CI failed on `topk_min_sort_none`, and the root cause is not found yet and its log is like;
> layer_test_utils.hpp:118 Relative comparison of values expected: 1.000000 and actual: 0.000000 at index 0 with threshold 0.01 failed" thrown in the test body.

1 is the expected value and the actual output value seems to be 0, and this issue needs further inspection.
In Ubuntu, correct value is generated, and this issue occurred on Windows only. So `topk_min_sort_none` tests are skipped for now, and will be inspected.

**Comment by steve-y** [comment]
Adding the following regex to skip_tests_config.cpp;
```
R"(.*ReferenceTopKTest.*aType=i8.*)",
R"(.*ReferenceTopKTest.*aType=i16.*)",
R"(.*ReferenceTopKTest.*aType=u8.*)",
R"(.*ReferenceTopKTest.*aType=u16.*)",
R"(.*ReferenceTopKTest.*aType=bf16.*)",
R"(.*ReferenceTopKTest.*aType=f64.*)",
```
because it generates the following log,
 > C++ exception with description "Interpreter backend doesn't implement evaluate method for OP TopK" thrown in the test body.

And as the log says, it seems the type i8, i16, u8, u16, bf16, f64 are not implemented yet, so this issue needs to be inspect more separately.

And the type i4, u4 generate Segmentation Fault when the testcases are generated and registered before the test start, and , so this issue needs to be inspect more separately as well.

File: docs/template_plugin/tests/functional/op_reference/topk.cpp [hunk 1]
```diff
@@ -0,0 +1,1189 @@
...
+class ReferenceTopKTest1d : public ReferenceTopKTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.A.data};
+        refOutData = {params.result0.data};
+    }
+
+    void SetUp1() {
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/topk.cpp:643]: Very strange name. Can we use more clear name?

File: docs/template_plugin/tests/functional/op_reference/topk.cpp [hunk 2]
```diff
@@ -0,0 +1,1189 @@
...
+    static std::shared_ptr<Function> CreateFunction1(const TopKParams& params) {
+        const auto A = std::make_shared<op::v0::Parameter>(params.A.type,
+                                                           params.A.shape);
+        const auto k = op::v0::Constant::create(params.k.type,
+                                                params.k.shape,
+                                                params.k.data.data());
+        const auto B = std::make_shared<op::v1::TopK>(A, k, params.axis, params.mode, params.sort);
+        const auto f = std::make_shared<Function>(OutputVector{B->output(1)}, ParameterVector{A});
+        return f;
+    }
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/topk.cpp:671]:
```suggestion
    static std::shared_ptr<Function> CreateFunction(const TopKParams& params, size_t out_idx) {
        const auto A = std::make_shared<op::v0::Parameter>(params.A.type,
                                                           params.A.shape);
        const auto k = op::v0::Constant::create(params.k.type,
                                                params.k.shape,
                                                params.k.data.data());
        const auto B = std::make_shared<op::v1::TopK>(A, k, params.axis, params.mode, params.sort);
        const auto f = std::make_shared<Function>(OutputVector{B->output(out_idx)}, ParameterVector{A});
        return f;
    }
```

File: docs/template_plugin/tests/functional/op_reference/topk.cpp [hunk 3]
```diff
@@ -0,0 +1,1189 @@
...
+        function = CreateFunction(params);
+        inputData = {params.A.data};
+        refOutData = {params.result0.data};
+    }
+
+    void SetUp1() {
+        auto params = GetParam();
+        function = CreateFunction1(params);
+        inputData = {params.A.data};
+        refOutData = {params.result1.data};
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/topk.cpp:647]: Can we add out index as a part of parameters and use don't use this method and custom pipeline

File: docs/template_plugin/tests/functional/op_reference/topk.cpp (hunk 1 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/topk.cpp]: It is updated to SetUpSecond.

File: docs/template_plugin/tests/functional/op_reference/topk.cpp (hunk 2 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/topk.cpp]: Update and commit with one CreateFunction.

File: docs/template_plugin/tests/functional/op_reference/topk.cpp (hunk 3 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/topk.cpp]: Add outIdx to parameter and don't call custom pipeline.

File: docs/template_plugin/tests/functional/op_reference/topk.cpp [hunk 4]
```diff
@@ -0,0 +1,1547 @@
...
+        testcaseName(testcaseName) {}
+
+    TopKParams(
+        const Tensor& A, const Tensor& k, const int64_t axis,
+        const op::v1::TopK::Mode mode, const op::v1::TopK::SortType sort,
+        const Tensor& result0, const Tensor& result1, size_t outIdx,
+        const std::string& testcaseName = "") :
+        A(A), k(k), axis(axis), mode(mode), sort(sort),
+        result0(result0), result1(result1), outIdx(outIdx),
+        testcaseName(testcaseName) {}
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/topk.cpp:32]:
```suggestion
    TopKParams(
        const Tensor& A, const Tensor& k, const int64_t axis,
        const op::v1::TopK::Mode mode, const op::v1::TopK::SortType sort,
        const Tensor& result0, const Tensor& result1, size_t outIdx = 0,
        const std::string& testcaseName = "") :
        A(A), k(k), axis(axis), mode(mode), sort(sort),
        result0(result0), result1(result1), outIdx(outIdx),
        testcaseName(testcaseName) {}
```

File: docs/template_plugin/tests/functional/op_reference/topk.cpp [hunk 5]
```diff
@@ -0,0 +1,1547 @@
...
+    op::v1::TopK::Mode mode;
+    op::v1::TopK::SortType sort;
+    Tensor result0;
+    Tensor result1;
+    size_t outIdx;
+    std::string testcaseName;
+};
+
+template <typename T>
+void copy_data(runtime::Tensor& t, const std::vector<T>& data) {
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/topk.cpp:46]: Why do we need this helper? Do we need this helper in other tests? If yes, should we move it to common utils?

Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/topk.cpp]: Removed. Thanks

File: docs/template_plugin/tests/functional/op_reference/topk.cpp (hunk 4 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/topk.cpp]: As the outIdx is not the last value, it does not seem to work as expected, so default value 0 to every outIdx. And it works. Thanks.

**Review by ilyachur — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Pull request merged**